metrics/github/cli.py

```diff
@@ -50,22 +50,24 @@ def is_open(pr, dt):
 
         return (closed - opened) >= threshold
 
-    with TimescaleDBWriter(GitHubPullRequests) as writer:
-        for monday in mondays:
-            dt = datetime.combine(monday, time(), tzinfo=UTC)
-            prs_open = [pr for pr in prs if is_open(pr, dt)]
-            prs_open = drop_archived_prs(prs_open, monday)
+    for monday in mondays:
+        dt = datetime.combine(monday, time(), tzinfo=UTC)
+        prs_open = [pr for pr in prs if is_open(pr, dt)]
+        prs_open = drop_archived_prs(prs_open, monday)
 
-            name = f"queue_older_than_{days_threshold}_days"
+        name = f"queue_older_than_{days_threshold}_days"
 
-            log.info(
-                "%s | %s | Processing %s PRs open at %s",
-                name,
-                org,
-                len(prs_open),
-                dt,
-            )
-            process_prs(writer, prs_open, monday, name=name)
+        log.info(
+            "%s | %s | Processing %s PRs open at %s",
+            name,
+            org,
+            len(prs_open),
+            dt,
+        )
+        processed_prs = list(iter_prs(prs_open, monday, name))
```

Comment:
If you stuff the return value of this function in to a list every time you call it, I think that's a hint that you've got the return type wrong.